dot/network/host.go

```diff
@@ -368,6 +369,53 @@ func (h *host) peers() []peer.ID {
 	return h.h.Network().Peers()
 }
 
+// addReservedPeers adds the peers `addrs` to the protected peers list and connects to them
+func (h *host) addReservedPeers(addrs ...string) error {
+	if len(addrs) < 1 {
+		return nil
+	}
+
+	for _, addr := range addrs {
+		maddr, err := ma.NewMultiaddr(addr)
+		if err != nil {
+			return err
+		}
+
+		addinfo, err := peer.AddrInfoFromP2pAddr(maddr)
+		if err != nil {
+			return err
+		}
+
+		h.h.ConnManager().Protect(addinfo.ID, "")
+		if err := h.connect(*addinfo); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// removeReservedPeers will remove the given peers from the protected peers list
+func (h *host) removeReservedPeers(ids ...string) error {
+	if len(ids) < 1 {
+		return nil
+	}
+
+	for _, id := range ids {
+		peerID, err := peer.Decode(id)
+		if err != nil {
+			return err
+		}
+
+		del := h.h.ConnManager().Unprotect(peerID, "")
+		if !del {
+			return errors.New("could not disconnect peer")
```

Comment:
Can you confirm if `Unprotect` also disconnects? 
I think `Unprotect` removes it from the map and returns false when the peer was not present in the protected map.

Reply:
Yes, `Unprotect` just removes the `peerID` from the map

Reply:
Good point, would be better to just jump to the next one instead of returning an error, I will fix that!